### ISSUE TITLE
Tiledata value where all jsonb values are null causes datatype.get_display_value to fail (7.5.x)

### ISSUE THREAD

**Issue by aj-he** [archesproject/arches#10829]
Resources that have tiles where all values are None are causing get_display_value to fail for nodes in them.

![image](https://github.com/archesproject/arches/assets/25295526/c9c310e8-1f1b-440c-8269-316d78ffd1bd)

![image](https://github.com/archesproject/arches/assets/25295526/73533f4f-8834-416f-b932-ccc5bb347611)

This appears to be caused by a previous bug fix for `BaseDataType.get_tile_data` that now causes the function to return None if only None values are found.

Commit: https://github.com/archesproject/arches/pull/10115/commits/e4a519e531aae9e31b3671a116fdfb72a6d8e62e

A number of the `datatype.get_display_value` functions do not handle being returned:

- concept
- concept-list
- date
- edtf
- file-list
- domain
- resource-intance
- node (handled but incorrect error message).

I am unsure if there are rules stopping these type of value empty tiles being created, but this issue has been made obvious having migrating an existing dataset

**Comment by jacobtylerwalls** [comment]
Hey @aj-he, I'm happy to look into this a bit further, but I need just a little help understanding the issue.

You mention `get_tile_data()` is returning None, but the first part of the condition checks `if data is not None`.

What is the failure that you are seeing?

> I am unsure if there are rules stopping these type of value empty tiles being created, but this issue has been made obvious having migrating an existing dataset

AFAICT this is intentional behavior as of 7.5 (#9074) -- all node k/v pairs are initialized to `[nodeid]: None`.

**Comment by aj-he** [comment]
Hey @jacobtylerwalls

So I have a particular tile that is passed in with tile["data"] value as ...
```json
{
  "a5419245-f121-11eb-be7b-a87eeabdefba": None,
  "a5419248-f121-11eb-86a9-a87eeabdefba": None,
  "a541e01e-f121-11eb-8ad7-a87eeabdefba": None
}
```
... which populates `data` in `get_tile_data`.

Then at line 196, it checks `if data not None`, which is true so moves on to...

 `and any(val is not None for val in data.values())` - here looks for at least one non-None value in order to `return data`, which is false for this tiledata.

My tile has only None values so moves on to the other tests and fails to meet any of their criteria. I put a print statement at the end to show it falls off the end and setting the target variable to None...

![image](https://github.com/archesproject/arches/assets/25295526/1a49c9e9-7b18-4bf7-9721-3c31df49daca)

```
# print out
Fell off the end of the get_tile_data function!!!!,

# tile
{'data': {'a5419245-f121-11eb-be7b-a87eeabdefba': None, 'a5419248-f121-11eb-86a9-a87eeabdefba': None, 'a541e01e-f121-11eb-8ad7-a87eeabdefba': None}, 'nodegroup_id': 'a541560c-f121-11eb-aa92-a87eeabdefba', 'parenttile_id': None, 'provisionaledits': None, 'resourceinstance_id': '7a856821-4361-4f85-b91e-f66f5b7bc129', 'sortorder': 0, 'tileid': 'a169c73c-9ffe-4685-a7b7-95ed7a4048f2'}

# data
{'a5419245-f121-11eb-be7b-a87eeabdefba': None, 'a5419248-f121-11eb-86a9-a87eeabdefba': None, 'a541e01e-f121-11eb-8ad7-a87eeabdefba': None} 

# provitionedits
None
```

![image](https://github.com/archesproject/arches/assets/25295526/2cedda18-e1fa-4dc0-9702-fec1e5c41598)

**Comment by aj-he** [comment]
Closed completed #10834
